plugins/modules/clusters.py

```diff
@@ -80,53 +90,95 @@
 def run_module():
 
     module_args = dict(
-        action=dict(type="str", required=False, default="list"),
+        state=dict(
+            type="str", required=False, choices=["absent", "present"], default=None
+        ),
         cluster_id=dict(type="str", required=False),
         # any API query parameters may have to be added here
         with_hosts=dict(type="bool", required=False, default=False),
+        name=dict(type="str", required=False),
+        openshift_version=dict(type="str", required=False),
+    )
+    token = os.environ.get("AI_API_TOKEN")
+    module = AnsibleModule(
+        argument_spec=module_args,
+        required_if=[
+            ("state", "present", ["name", "openshift_version"]),
+            ("state", "absent", ["cluster_id"]),
+        ],
+        supports_check_mode=False,
     )
-    token = os.environ.get('AI_API_TOKEN')
-    module = AnsibleModule(argument_spec=module_args, supports_check_mode=False)
 
     # Fail if requests is not installed
     if not HAS_REQUESTS:
-        module.fail_json(msg=missing_required_lib('requests'), exception=REQUESTS_IMPORT_ERROR)
+        module.fail_json(
+            msg=missing_required_lib("requests"), exception=REQUESTS_IMPORT_ERROR
+        )
 
     # Set headers
-    headers = {
-        'Content-Type': 'application/json',
-        'Authorization': f'Bearer {token}'
-    }
+    headers = {"Content-Type": "application/json", "Authorization": f"Bearer {token}"}
+
+    # Delete cluster
+    if module.params.get("state") == "absent":
+        if not module.params.get("cluster_id"):
```

Comment:
This test should not be required since the `required_if` module argument will catch it.

Reply:
Good catch, updated